action/protocol/staking/candidate_statemanager.go

```diff
@@ -51,7 +51,7 @@ type (
 )
 
 // NewCandidateStateManager returns a new CandidateStateManager instance
-func NewCandidateStateManager(sm protocol.StateManager) (CandidateStateManager, error) {
+func NewCandidateStateManager(sm protocol.StateManager, enableSMStorage bool) (CandidateStateManager, error) {
```

Comment:
this should not be an input to NewCandidateStateManager()
it is only specific for BucketPool

Reply:
why?